src/action.c

```diff
@@ -34,24 +35,38 @@ lf_ret_t Action_schedule(Action *self, interval_t offset, const void *value) {
   Environment *env = self->super.parent->env;
   Scheduler *sched = &env->scheduler;
   void *payload = NULL;
-  validate(value);
 
   env->enter_critical_section(env);
 
   // Dont accept events before we have started
   // TODO: Do we instead need some flag to signal that we have started?
   if (sched->start_time == NEVER) {
-    LF_ERR(TRIG, "Action %p cannot schedule events before start tag", self);
     env->leave_critical_section(env);
+    LF_ERR(TRIG, "Action %p cannot schedule events before start tag", self);
     return LF_ERR;
   }
 
-  ret = self->super.payload_pool->allocate(self->super.payload_pool, &payload);
-  if (ret != LF_OK) {
-    return ret;
+  if (self->super.payload_pool->capacity == 0 && value != NULL) {
+    // user tried to schedule a action that does not have any value
+    env->leave_critical_section(env);
+    LF_ERR(TRIG, "Scheduling an value, that doesn't have value!");
+    sched->do_shutdown(sched, sched->current_tag);
```

Comment:
I would say that it is the callers job of calling `do_shutdown`, should it instead be done in the `lf_schedule` macro?  ideally it would be in the `throw` macro, but it then needs a pointer to the environment...